pkg/storage/snapshot/snapshot.go

```diff
@@ -92,8 +92,7 @@ func vmSnapshotSucceeded(vmSnapshot *snapshotv1.VirtualMachineSnapshot) bool {
 }
 
 func vmSnapshotProgressing(vmSnapshot *snapshotv1.VirtualMachineSnapshot) bool {
-	return vmSnapshotError(vmSnapshot) == nil && !VmSnapshotReady(vmSnapshot) &&
```

Comment:
With the error being transient and possibly to have one and then it is gone the vmsnapshot can still be in progress and have error != nil.
`vmSnapshotProgressing` is used in several places for example `canUnlockSource` and I dont think we need to allow unlock source if error != nil, we should if the phase is actually different then in progress.
In this check in case of error we will return false: https://github.com/kubevirt/kubevirt/blob/02621d08b868c50288e803858936e7cf4fba3400/pkg/storage/snapshot/snapshot.go#L175
and then in the else if can unlock we will return true: https://github.com/kubevirt/kubevirt/blob/02621d08b868c50288e803858936e7cf4fba3400/pkg/storage/snapshot/snapshot.go#L195
so basically I want to continue vmsnapshot and stay in progress in case of error.